tools/gen_allocs.c

```diff
@@ -80,13 +80,13 @@ int
 gen_alloc2 ( FILE * fp , char * structname , char * structname2 , node_t * node, int *j, int *iguy, int *fraction, int numguys, int frac, int sw ) /* 1 = allocate, 2 = just count */
 {
   node_t * p ;
-  int tag ;
+  int tag = 0 ;
   char post[NAMELEN], post_for_count[NAMELEN] ;
-  char fname[NAMELEN], dname[NAMELEN], dname_tmp[NAMELEN] ;
-  char x[NAMELEN] ;
-  char x2[NAMELEN], fname2[NAMELEN] ;
-  char dimname[3][NAMELEN] ;
-  char tchar ;
+  char fname[NAMELEN], dname[NAMELEN + 11], dname_tmp[NAMELEN] ;
```

Comment:
I'm guessing a lot of these extra `NAMELEN` additions are due to adjusting for fields within the registry that have longer lengths than first anticipated. Would it be possible to make these all a common size, or if not for those that share a common reason for increased size to break it out into an aptly named `#define` macros?

Reply:
I think it's mostly concatenating variables of length `NAMELEN` or putting extra bits on the end, e.g. `sprintf(dname, "%s/external/io_netcdf", fname);`, though that specific example would require `char dname[NAMELEN + 19];`.  How much extra room the destination buffer requires depends on how much extra stuff gets tacked on, though I could define  `BUFFER_FOR_DESTINATION` to be the largest such requirement and use `NAMELEN + BUFFER_FOR_DESTINATION` for most destination buffers or `NAMELEN + 2 * BUFFER_FOR_DESTINATION` for others.

Reply:
That sounds like a good path forward. I'm mostly angling for avoiding cryptic magic numbers where it might not be obvious why that number is what it is, especially when compared to similarly named and initialized variables.

Reply:
Changed, and most warnings eliminated again.  There is one left, which I don't understand and can't get to go away:
```
gen_scalar_indices.c: In function ‘gen_scalar_indices1.isra’:
gen_scalar_indices.c:194:57: error: ‘%s’ directive output between 0 and 124999 bytes may exceed minimum required size of 4095 [-Werror=format-truncation=]
  194 |                   snprintf(fourd_bnd, NAMELEN_LONG + 3, "%s_b",assoc_4d) ;
      |                                                         ^      ~
gen_scalar_indices.c:194:19: note: ‘snprintf’ output between 3 and 125002 bytes into a destination of size 125003
  194 |                   snprintf(fourd_bnd, NAMELEN_LONG + 3, "%s_b",assoc_4d) ;
      |                   ^
```
but I think that will continue to be as fine as it was; it has enough space.